File: libgomp/testsuite/libgomp.fortran/metadirective-2.f90
! { dg-do run }

program test
  implicit none
  integer, parameter :: N = 100
  real, parameter :: PI_CONST = 2.0*acos(0.0)
  real, parameter :: E_CONST = exp(1.0)
  real, parameter :: EPSILON = 0.001
  integer :: i
  real :: a(N)

  !$omp target map(from: a)
    call f (a, PI_CONST)
  !$omp end target

  do i = 1, N
    if (abs (a(i) - (PI_CONST * i)) .gt. EPSILON) stop 1
  end do

  call f (a, E_CONST)

  do i = 1, N
    if (abs (a(i) - (E_CONST * i)) .gt. EPSILON) stop 2
  end do
contains
  subroutine f (a, x)
    integer :: i
    real :: a(N), x
    !$omp declare target

    !$omp metadirective &
    !$omp&  when (construct={target}: distribute parallel do ) &
    !$omp&  default(parallel do simd)
      do i = 1, N
	a(i) = x * i
      end do
  end subroutine
end program

File: libgomp/testsuite/libgomp.c-c++-common/metadirective-2.c
/* { dg-do run } */

#include <math.h>

#define N 100
#define EPSILON 0.001

#pragma omp declare target
void
f (double a[], double x) {
  int i;

  #pragma omp metadirective \
	when (construct={target}: distribute parallel for) \
	default (parallel for simd)
   for (i = 0; i < N; i++)
     a[i] = x * i;
}
#pragma omp end declare target

int
main (void)
{
  double a[N];
  int i;

  #pragma omp target teams map(from: a[0:N])
    f (a, M_PI);

  for (i = 0; i < N; i++)
    if (fabs (a[i] - (M_PI * i)) > EPSILON)
      return 1;

  f (a, M_E);

  for (i = 0; i < N; i++)
    if (fabs (a[i] - (M_E * i)) > EPSILON)
      return 1;

  return 0;
 }
